airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -186,6 +186,42 @@ def post(
         response = self._session.post(self.BASE_URL + url, params=params, json=data)
         return self._parse_and_handle_errors(response), response
 
+    def get_custom_object_schemas(self) -> Mapping[str, Any]:
+        data, response = self.get("/crm/v3/schemas", {})
+        schemas = {}
+        if response.ok and "results" in data:
+            for raw_schema in data["results"]:
+                schema = self.generate_schema(raw_schema=raw_schema)
+                schemas[raw_schema["name"]] = schema
+        return schemas
+
+    def generate_schema(self, raw_schema: Mapping[str, Any]) -> Mapping[str, Any]:
+        properties = {
+            "id": {"type": ["null", "string"]},
+            "createdAt": {"type": ["null", "string"], "format": "date-time"},
+            "updatedAt": {"type": ["null", "string"], "format": "date-time"},
+            "archived": {"type": ["null", "boolean"]},
+            "properties": {}
+        }
+        schema = {"$schema": "http://json-schema.org/draft-07/schema#", "type": "object", "additionalProperties": True, "properties": properties}
+        for field in raw_schema["properties"]:
+            schema["properties"]["properties"][field["name"]] = self._field_to_property_schema(field)
+        return schema
+
+    @staticmethod
+    def _field_to_property_schema(field_params: Mapping[str, Any]) -> Mapping[str, Any]:
+        type = field_params["type"]
+        property_schema = {}
+        if type == "enumeration" or type == "string":
+            property_schema = {"type": ["null", "string"]}
+        elif type == "datetime":
+            property_schema = {"type": ["null", "date-time"]}
+        elif type == "date":
+            property_schema = {"type": ["null", "date"]}
+        elif type == "number":
+            property_schema = {"type": ["null", "number"]}
+        return property_schema
```

Comment:
I am not sure if it is important, but the schema can be empty here (e.g. on `bool`)